packages/api/tests/lambdas/test-write-db-dlq-records-to-s3.js

```diff
@@ -103,3 +107,89 @@ test.serial('write-db-dlq-records-to-s3 throws error if system bucket is not def
     { message: 'System bucket env var is required.' }
   );
 });
+
+test('determineExecutionName returns execution name if it exists in cumulus_meta', (t) => {
+  const executionName = 'someExecutionName';
+  t.is(determineExecutionName({
+    cumulus_meta: {
+      execution_name: executionName,
+    },
+  }),
+  executionName);
+});
+
+test('determineExecutionName returns "unknown" if getExecutionName throws error', (t) => {
+  t.is(determineExecutionName({}), 'unknown');
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an SQS message', (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    eventSource: 'aws:sqs',
+    body: JSON.stringify(cumulusMessage),
+  };
+  t.deepEqual(unwrapDeadLetterCumulusMessage(testMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS States message', (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    source: 'aws.states',
+    detail: {
+      output: JSON.stringify(cumulusMessage),
+    },
+  };
+  t.deepEqual(unwrapDeadLetterCumulusMessage(testMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS States message with only input', (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    source: 'aws.states',
+    detail: {
+      input: JSON.stringify(cumulusMessage),
+    },
+  };
+  t.deepEqual(unwrapDeadLetterCumulusMessage(testMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS states message within an SQS message', (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testStatesMessage = {
+    source: 'aws.states',
+    detail: {
+      output: JSON.stringify(cumulusMessage),
+    },
+  };
+  const testSqsMessage = {
+    sourceEvent: 'aws:sqs',
+    body: JSON.stringify(testStatesMessage),
+  };
+  t.deepEqual(unwrapDeadLetterCumulusMessage(testSqsMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage returns an non-unwrappable message', (t) => {
```

Comment:
🐜 
```suggestion
test('unwrapDeadLetterCumulusMessage returns input as-is if it cannot be unwrapped', (t) => {
```
or something like that? 🤔